src/odemis/gui/model/__init__.py

```diff
@@ -1562,23 +1577,70 @@ def moveStageTo(self, pos):
 
         move = {"x": pos[0], "y": pos[1]}
 
-        # clip to the range of the axes
-        for ax, p in move.items():
-            axis_def = self._stage.axes[ax]
-            if (hasattr(axis_def, "range") and
-                not axis_def.range[0] <= p <= axis_def.range[1]
-               ):
-                p = max(axis_def.range[0], min(p, axis_def.range[1]))
-                move[ax] = p
-                logging.info("Restricting stage axis %s move to %g mm due to stage limit",
-                             ax, p * 1e3)
+        # If needed clip current movements in x/y direction to the maximum allowed stage limits
+        stage_limits = self._getStageLimitsXY()
+        if "x" in stage_limits and not stage_limits["x"][0] <= move["x"] <= stage_limits["x"][1]:
+            move["x"] = max(stage_limits["x"][0], min(move["x"], stage_limits["x"][1]))
+            logging.info("Movements of the stage in x direction are limited by the stage limit, the movement is "
+                         "restricted to %s mm." % (move["x"] * 1e3))
```

Comment:
You can make the log a little less verbose, and still say more, like:
`"Movements of the stage in x limited to %s, restricting move to %s mm." % (stage_limits["x"], move["x"] * 1e3)`